web_external/js/models/SourceModel.js

```diff
@@ -0,0 +1,2 @@
+minerva.models.SourceModel = minerva.models.MinervaModel.extend({
```

Comment:
I could try it that way.  The subclasses would need to pass up the params and the path.

What I'm worried about is having a single method in the Source Model that will need to change as we add new Source types, given the fact that we don't have front-end testing, I'd be worried about possible breakage.  My preference for now is to build out the Source Model classes a bit, and then once we have some front-end testing in place do a unifying refactor.

I could do as you suggest though, I'm really on the fence and could be swayed by a strong argument.